pandas/tests/io/test_pickle.py

```diff
@@ -635,3 +635,13 @@ def test_pickle_big_dataframe_compression(protocol, compression):
         partial(pd.read_pickle, compression=compression),
     )
     tm.assert_frame_equal(df, result)
+
+
+def test_pickle_frame_v124_unpickle_130():
+    # GH#42345 DataFrame created in 1.2.x, unpickle in 1.3.x
+    path = os.path.join(legacy_dirname, "1.2.4", "empty_frame_v1_2_4-GH#42345.pkl")
+    with open(path, "rb") as fd:
+        df = pickle.load(fd)
+
+    expected = pd.DataFrame()
```

Comment:
can i add another case with columns and index (respectively) as well (but no rows)

Reply:
this test is highly focused, let's hold off on adding to this test and do a dedicated pass using the generate_legacy_storage_files pattern

Reply:
ahh ok, yeah i think that would be a great idea.